src/NavbarToggle.js

```diff
@@ -15,6 +18,14 @@ let NavbarToggle = React.createClass({
     $bs_navbar_onToggle: PropTypes.func,
   },
 
+  getInitialState() {
+    return {collapsed: true};
+  },
+
+  handleCollapsedState() {
+    this.setState({collapsed: !this.state.collapsed});
```

Comment:
so the potential problem with this approach is that we are tracking the collapsed state in two places now, this Toggle and the parent Navbar, they could get out of sync, or the API change, etc.
I'd be better, since the Navbar is already exposing `expanded` on context to just use that value instead of the additional state logic here.